lib/calendar.ts

```diff
@@ -2077,6 +2156,54 @@ abstract class ChineseBaseHelper extends HelperBase {
   maximumMonthLength(/* calendarDate */) {
     return 30;
   }
+  maxLengthOfMonthCodeInAnyYear(monthCode: string) {
+    // See note below about ICU4C vs ICU4X. It is possible this override should
+    // always return 30.
+    if (
+      monthCode === 'M01L' ||
+      monthCode === 'M09L' ||
+      monthCode === 'M10L' ||
+      monthCode === 'M11L' ||
+      monthCode === 'M12L'
+    ) {
+      return 29;
+    }
+    return 30;
+  }
+  override monthDaySearchStartYear(monthCode: string, day: number) {
+    // Note that ICU4C actually has _no_ years in which leap months M01L and
+    // M09L through M12L have 30 days. The values marked with (*) here are years
+    // in which the leap month occurs with 29 days. ICU4C disagrees with ICU4X
+    // here and it is not clear which is correct.
+    switch (monthCode) {
+      case 'M01L':
+        return 1651; // *
+      case 'M02L':
+        return day < 30 ? 1947 : 1765;
+      case 'M03L':
+        return day < 30 ? 1966 : 1955;
+      case 'M04L':
+        return day < 30 ? 1963 : 1944;
+      case 'M05L':
+        return day < 30 ? 1971 : 1952;
+      case 'M06L':
+        return day < 30 ? 1960 : 1941;
+      case 'M07L':
+        return day < 30 ? 1968 : 1938;
+      case 'M08L':
+        return day < 30 ? 1957 : 1718;
+      case 'M09L':
+        return 1832; // *
+      case 'M10L':
+        return 1870; // *
+      case 'M11L':
+        return 1814; // *
+      case 'M12L':
+        return 1890; // *
+      default:
+        return 1972;
+    }
```

Comment:
Maybe for post-rebase: 

```suggestion
    const monthMap: Record<string, [number, number]> = {
      M01L: [1651, 1651], // *
      M02L: [1947, 1765],
      M03L: [1966, 1955],
      M04L: [1963, 1944],
      M05L: [1971, 1952],
      M06L: [1960, 1941],
      M07L: [1968, 1938],
      M08L: [1957, 1718],
      M09L: [1832, 1832], // *
      M10L: [1870, 1870], // *
      M11L: [1814, 1814], // *
      M12L: [1890, 1890], // *
   };
   const years = monthMap[monthCode] ?? [1972, 1972]
   return day < 30 ? years[0] : years[1];
```